src/main/webapp/app/shared/components/button.component.ts

```diff
@@ -24,7 +25,13 @@ export enum ButtonSize {
 @Component({
     selector: 'jhi-button',
     template: `
-        <button [ngClass]="['jhi-btn', 'btn', btnType, btnSize]" ngbTooltip="{{ tooltip | translate }}" [disabled]="disabled || isLoading" (click)="onClick.emit($event)">
+        <button
```

Comment:
I also thought about this - one way to solve this would be using a HostBinding.
This way the `btn` style would not be inside the tag, but directly on it. Therefore also the `btn-group` should work with the component.

E.g. 
```
@HostBinding('class') get classes(): string {
  return classNames(this.getDynamicClassName(), {
    'is-open': this.isOpen,
    'has-children': this.hasChildren
  });
}
```

This would then of course have the disadvantage of constant evaluation.

Reply:
This PR would btw also finish the implementation of the issue: https://github.com/ls1intum/Artemis/pull/1013